tests/functional/object/mcg/test_noobaa_db_cleartext_postgres_password.py

```diff
@@ -43,3 +44,48 @@ def test_noobaa_db_cleartext_postgres_password(self):
         assert (
             "set=password" not in nooobaa_db_pod_logs
         ), f"noobaa-db pod logs include password logs:{nooobaa_db_pod_logs}"
+
+    @bugzilla("2240778")
+    @polarion_id("OCS-6183")
+    def test_nb_db_password_in_core_and_endpoint(self):
+        """
+        Verify that postgres password is not exposed in
+        noobaa core and endpoint logs
+
+        1. Get the noobaa core log
+        2. Get the noobaa endpoint log
+        3. Verify postgres password doesnt exist in the endpoint and core logs
+
+        """
+        # get the noobaa db password
+        _, noobaa_db_password = get_noobaa_db_credentials_from_secret()
+
+        # get noobaa core log and verify that the password is not
+        # present in the log
+        nooba_core_pod = pod.get_noobaa_core_pod()
+        noobaa_core_pod_logs = pod.get_pod_logs(nooba_core_pod.name)
+        filtered_log = filter_pod_logs(
```

Comment:
https://github.com/red-hat-storage/ocs-ci/blob/e90b2edea045f472a52ceab4609860963699ead4/ocs_ci/helpers/helpers.py#L4562 this existing fun doesn't help?

Reply:
verify_log_exist_in_pods_logs() only checks for a single piece of log in whole log. It cant look for pattern of log. With filter_pod_logs, i will be able to filter the pod logs by certain pattern of logs.

Reply:
What about [search_pattern_in_pod_logs](https://github.com/red-hat-storage/ocs-ci/blob/e90b2edea045f472a52ceab4609860963699ead4/ocs_ci/ocs/resources/pod.py#L3528)? It searches for the given regular expression pattern in the logs of a pod